venues/ICLR.cc/2019/Conference/python/review-assignment-stage.py

```diff
@@ -0,0 +1,103 @@
+'''
+Reviewer Assignment Stage:
+- Bidding Task / interface is closed.
+- Initial paper-reviewer and paper-areachair matches are computed.
+- Program chairs browse match using OpenReview Matching System and make hand-adjustments as needed.
+- Paper-reviewer and paper-areachair assignments are finalized and deployed.
+
+'''
+
+import argparse
+import openreview
+import iclr19
+import invitations
+import notes
+import time
+import json
+
+import matcher
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--baseurl', help="base url")
+    parser.add_argument('--username')
+    parser.add_argument('--password')
+    args = parser.parse_args()
+
+    client = openreview.Client(baseurl=args.baseurl, username=args.username, password=args.password)
+
+    blind_submissions = openreview.tools.iterget_notes(client, invitation=iclr19.BLIND_SUBMISSION_ID)
+
+    # At this point, all reviewers should have been
+    # converted to profile IDs and deduplicated.
+    reviewers_group = client.get_group(iclr19.REVIEWERS_ID)
+    assert all(['~' in member for member in reviewers_group.members]), 'not all reviewers have been converted to profile IDs'
+    reviewer_profiles = client.get_profiles(reviewers_group.members)
+
+    invitations.disable_bids(client)
+
+    # create metadata
+    metadata_notes = []
+    metadata_inv = client.post_invitation(iclr19.metadata_inv)
+    for blind_note in blind_submissions:
+        original_note = client.get_note(id=blind_note.original)
+        paper_bids = client.get_tags(invitation=iclr19.add_bid.id)
+        new_metadata_note = notes.post_metadata_note(client, blind_note, original_note, reviewer_profiles, paper_bids)
+        metadata_notes.append(new_metadata_note)
+
+    with open('../data/iclr19-match-config-example.json') as f:
+        config = json.load(f)
+
+    # This could be set by hand if reviewers or papers have specific supplies/demands
+    supplies = [config['max_papers']] * len(reviewer_profiles)
+    demands = [config['max_users']] * len(metadata_notes)
+
+    # run initial match
+    encoder = matcher.metadata.Encoder(
+        metadata_notes,
+        config,
+        [r.id for r in reviewer_profiles])
+
+    flow_solver = matcher.Solver(supplies, demands, encoder.cost_matrix, encoder.constraint_matrix)
+    solution = flow_solver.solve()
```

Comment:
I don't know why you are doing the solver here, it is just for the demo?

Reply:
no, we need this to initialize the matching browser based on TPMS scores and bid scores. we don't have scores right now, so there is dummy data. but this script is supposed to generate assignments with the solver.

Reply:
I think I understand what you're saying now - are you wondering why we don't do the matching in a separate script?

I'm not sure. maybe we should. let's discuss it

Reply:
This code is in the matching repo, why don’t you run the matching from there? 

You have to create only the metadata notes here, right?